added 16311 :)

harshul-s opened the pull request:
# Summary of Changes

- Please indicate the classes you are modifying:
  - added 16-311 under electives and updated readme

# Checklist:

Make sure that your contributions:

- [ ] Are accurate and will not mislead the reader
- [ ] Do not encourage or cause academic integrity violations
- [ ] Do not offend professors or unfairly criticize the university

harshul-s committed:
Create 16311.md

added 16-311

harshul-s committed:
Update README.md

added 16311 to readme

harshul-s committed:
Update 16311.md

fixed typos

mikinty commented on electives/16311.md:
```diff
@@ -0,0 +1,59 @@
+- **Start Labs Early!!!!!**
+- Attend Lecture/ Closely review the lecture slides
+- Being comfortable with matlab definitely helps with early parts of the homeworks and labs
+- while this class does coreq 15-122 and stresses C experience, the majority of the programming you do is in Python, robot C appears to no longer be used to program your robot
```
This should be capitalized but I'm not going to nitpick too much

mikinty commented on electives/16311.md:
```diff
@@ -0,0 +1,59 @@
+- Attend Lecture/ Closely review the lecture slides
+- Being comfortable with matlab definitely helps with early parts of the homeworks and labs
+- while this class does coreq 15-122 and stresses C experience, the majority of the programming you do is in Python, robot C appears to no longer be used to program your robot
+- TA office hours particularily Lab OH to get advice on design decisions or debug your code is super valuable.
```
"Particularly" is the correct spelling here

I think you're missing a comma 
```suggestion
- TA office hours, particularity Lab OH, are valuable to get advice on design decisions or to debug your code.
```

mikinty commented on electives/16311.md:
```diff
@@ -0,0 +1,59 @@
+- while this class does coreq 15-122 and stresses C experience, the majority of the programming you do is in Python, robot C appears to no longer be used to program your robot
+- TA office hours particularily Lab OH to get advice on design decisions or debug your code is super valuable.
+- Piazza is pretty helpful for clarifying things on the Lab handout/Homeworks since they can initially contain some ambiguity that gets clarified when you/others ask.
+- The difficulty of this class can scale quite significantly if you have taken previous courses as part of the RI major/minor. If you have prior experience from those classes the content and labs will likely be quite familiar to you. This class is the first one I have taken in robotics and I have needed to self-study some concepts that may not be exhaustively explained in lecture.
```
```suggestion
- The difficulty of this class can vary quite significantly if you have taken previous courses as part of the RI major/minor. If you have prior experience from those classes, the content and labs will likely be quite familiar to you. This class is the first one I have taken in robotics and I have needed to self-study some concepts that may not be exhaustively explained in lecture.
```

mikinty commented on electives/16311.md:
```diff
@@ -0,0 +1,59 @@
+
+# Labs
+
+44% of your grade is tied to the labs so expect this part of the class to occupy the lion's share of the time you dedicate to it. Most Labs are completed self selected groups sized 2-3 but some are individual/pair based. I would strongly reccomend forming a group of 3. You are encouraged to have students from different majors ECE/CS/Mech-E to share your expertise in different area's but it's not  ...
```
```suggestion
44% of your grade is tied to the labs so expect this part of the class to occupy the lion's share of the time you dedicate to it. Most Labs are completed in self-selected groups sized 2-3, but some are individual/pair based. I would strongly reccomend forming a group of 3. You are encouraged to have students from different majors ECE/CS/Mech-E to share your expertise in different areas, but it's not required. Labs typically span a week, with a demo period on Tuesday. They can vary quite significantly in difficulty, but leveraging the help from TAs and starting early makes them manageable. 
```

mikinty commented on electives/16311.md:
```diff
@@ -0,0 +1,59 @@
+
+# Homeworks
+
+The homework component typically has a written part designed to cover lecture content/theory and then occasionally an accompanying programming element to apply what you have learned e.g Training a NN in pyTorch when learning about Reinforcement Learning. They aren't designed to be super challenging and are due on Wednesday the day after labs so it's definitely feasible to Start/Finish them in a d ...
```
```suggestion
The homework component typically has a written part designed to cover lecture content/theory and then occasionally an accompanying programming element to apply what you have learned e.g Training a NN in PyTorch when learning about Reinforcement Learning. They aren't designed to be super challenging and are due on Wednesday the day after labs, so it's definitely feasible to Start/Finish them in a day.
```

mikinty reviewed (changes requested):
Overall great, added mostly grammar feedback.

harshul-s committed:
applied feedback

Co-authored-by: Michael You <youmichaelc@gmail.com>

harshul-s committed:
Update 16311.md

harshul-s commented:
I think I forgot to hit commit  when I proofread it, added your suggestions to a batch commit and fixed the capitalization on the "while" for good measure. Thank you for your time and feedback!

mikinty commented on electives/16311.md:
```diff
@@ -0,0 +1,59 @@
+- Attend Lecture/ Closely review the lecture slides
+- Being comfortable with matlab definitely helps with early parts of the homeworks and labs
+- While this class does coreq 15-122 and stresses C experience, the majority of the programming you do is in Python, robot C appears to no longer be used to program your robot
+- TA office hours, particularity Lab OH, are valuable to get advice on design decisions or to debug your code.
```
```suggestion
- TA office hours, particularly Lab OH, are valuable to get advice on design decisions or to debug your code.
```

mikinty commented on electives/16311.md:
```diff
@@ -0,0 +1,59 @@
+
+44% of your grade is tied to the labs so expect this part of the class to occupy the lion's share of the time you dedicate to it. Most Labs are completed in self-selected groups sized 2-3, but some are individual/pair based. I would strongly reccomend forming a group of 3. You are encouraged to have students from different majors ECE/CS/Mech-E to share your expertise in different areas, but it's  ...
+
+Since demos are on Tuesday, I personally would suggest starting the labs at the latest on Saturday. The labs typically have 2-3 components tied to them. Designing a physical robot out of the lego BrickPi Kit, Programming and sometimes tuning constants through experimentaton/trial and error.
```
```suggestion
Since demos are on Tuesday, I personally would suggest starting the labs at the latest on Saturday. The labs typically have 2-3 components tied to them. Designing a physical robot out of the lego BrickPi Kit, Programming and sometimes tuning constants through experimentation/trial and error.
```

mikinty commented on electives/16311.md:
```diff
@@ -0,0 +1,59 @@
+- **Start Labs Early!!!!!**
+- Attend Lecture/ Closely review the lecture slides
+- Being comfortable with matlab definitely helps with early parts of the homeworks and labs
+- While this class does coreq 15-122 and stresses C experience, the majority of the programming you do is in Python, robot C appears to no longer be used to program your robot
```
```suggestion
- While this class does coreq 15-122 and stresses C experience, the majority of the programming you do is in Python. Robot C appears to no longer be used to program your robot.
```

mikinty replied:
What is robot c?

harshul-s replied:
I believe it's a programming language syntactically similar to C. [link here](http://cmra.rec.ri.cmu.edu/products/teaching_robotc_tetrix/fundamentals/introtoprogramming/documents/Fundamentals_ROBOTC.pdf) which is what I think prompted the C prerequisite. However, the current setup for the bots supports python which is what we're using instead. For the sake of concision should just remove that side-note since it doesn't have any bearing on the current course?

mikinty replied:
Yup don't include it if it doesn't apply to the current course. 

If you would like, you can include a footnote about the old dependency, but it's not necessary.

harshul-s replied:
removed it!

harshul-s committed:
Update 16311.md

Removed unnecessary aside on Robot C.

harshul-s committed:
Apply suggestions from code review

Co-authored-by: Michael You <youmichaelc@gmail.com>

mikinty reviewed (approved):
Thank you for working with me on this review, and thanks for the contribution!
